### DISCUSSION TITLE
hideIf() in the previous version

### DISCUSSION THREAD

**Discussion by 22syamsudar** [rappasoft/laravel-livewire-tables#252] [Q&A]
I was using hideIf() in the previous version of this package... 
how is in this version ? thanks guys.!

**Comment by rappasoft** [comment]
I'm not sure it would work in this version. It would work in the case where the cells are generated from the Column class, but in the case where rowView is supplied, I would only be able to hide the header and not the rest of the column because I don't have access to the columns when iterating the cells from the rowView. Unless I'm missing something or anyone has another suggestion on how to do it. Or it might just be a feature available if you don't use rowView, but I'm not a fan of that approach.

**Reply by 22syamsudar** [reply]
yeah. i really like that hideIf() to combine with spatie permission that i use.. and its really nice..
but in the new version i can't anymore..
i just  try to call the spatie function inside the cell. but can't hide the column...  :(

**Reply by rappasoft** [reply]
Even if I added it, it wouldn't be able to be used with custom row view for that reason, you don't know what column your're in.

**Reply by 22syamsudar** [reply]
it's ok...
for now i will make condition to display different view accourding to the permission... 
thanks for your response above...

**Reply by rappasoft** [reply]
I can probably add it to work with the column builder, and when it comes to the rowView, you'd just have to wrap each cell in the same conditional statement. That would probably work I'll just have to document it well.

**Reply by 22syamsudar** [reply]
nice...
thanks for the suggestion... 
i check it out.......
we hope everybody will see this conversation to solve the same problem..

**Reply by rappasoft** [reply]
That is not an option in this package, however if you don't use pagination, you do have access to $index inside rowView which literally is what you want starting at zero. But I wouldn't consider that a solution, it's just a byproduct of how I wrote the loop.

**Reply by rappasoft** [reply]
Ok I added it to develop: https://github.com/rappasoft/laravel-livewire-tables/commit/77993df6a9d6c2f66ec3b358ae0c4bdac742940d You can use that branch if you need it right now.

This will be added to the documentation on the next release:

If you would like to show/hide columns based on a conditional, you may use the `hideIf` method on the Column builder:

```php
Column::make('Special Field')
    ->hideIf(! auth()->user()->isAdmin());
```

**Note:** This only works for the corresponding cells if using the column builder to also build the rest of the table.

## When using `rowView` to render your cells:

Since rowView does not keep track of the column loop, you must also wrap the same conditional around the cells when using this method:

```html
@if (! auth()->user()->isAdmin())
    <x-livewire-tables::table.cell>
        {{ $row->special_field ?? 'N/A' }}
    </x-livewire-tables::table.cell>
@endif
```

**Reply by 22syamsudar** [reply]
really nice..
we appreciate you man.. thanks

**Comment by lrljoe** [comment]
This is present in v2+
